javalib/src/main/scala/java/math/BigInteger.scala

```diff
@@ -0,0 +1,1500 @@
+        }
+        _n = 1 << (_n & 31)
+        (digit & _n) != 0
+    }
+  }
+
+  /**
+   * Returns the two's complement representation of this BigInteger in a byte
+   * array.
+   *
+   * @return two's complement representation of {@code this}.
+   */
+  /* Todo: This is the original version of the above method, copied here for refence in CR - Needs deleting when CR done
+
+   def toByteArray(): Array[Byte] = {
+   if (this.sign == 0) {
+     return Array[Byte](0)
+   }
+   val temp: BigInteger = this
+   val bitLen = bitLength()
+   val iThis = getFirstNonzeroDigit
+   var bytesLen = (bitLen >> 3) + 1
+   /*
+    * Puts the little-endian int array representing the magnitude of this
+    * BigInteger into the big-endian byte array.
+    */
+   val bytes = Array.ofDim[Byte](bytesLen)
+   var firstByteNumber = 0
+   var highBytes: Int = 0
+   var digitIndex = 0
+   var bytesInInteger = 4
+   var digit: Int = 0
+   var hB: Int = 0
+
+   if (bytesLen - (numberLength << 2) == 1) {
+     bytes(0) = (if ((sign < 0)) -1 else 0).toByte
+     highBytes = 4
+     firstByteNumber += 1
+   } else {
+     hB = bytesLen & 3
+     highBytes = if ((hB == 0)) 4 else hB
+   }
+   digitIndex = iThis
+   bytesLen -= iThis << 2
+   if (sign < 0) {
+     digit = -temp.digits(digitIndex)
+     digitIndex += 1
+     if (digitIndex == numberLength) {
+       bytesInInteger = highBytes
+     }
+     var i = 0
+     while (i < bytesInInteger) {
+       bytesLen -=1
+       bytes(bytesLen) = digit.toByte
+       i += 1
+       digit >>= 8
+     }
+     while (bytesLen > firstByteNumber) {
+       digit = ~temp.digits(digitIndex)
+       digitIndex += 1
+       if (digitIndex == numberLength) {
+         bytesInInteger = highBytes
+       }
+       var i = 0
+       while (i < bytesInInteger) {
+         bytesLen -=1
+         bytes(bytesLen) = digit.toByte
+         i += 1
+         digit >>= 8
+       }
+     }
+   } else {
+     while (bytesLen > firstByteNumber) {
+       digit = temp.digits(digitIndex)
+       digitIndex += 1
+       if (digitIndex == numberLength) {
+         bytesInInteger = highBytes
+       }
+       var i = 0
+       while (i < bytesInInteger) {
+         bytesLen -=1
+         bytes(bytesLen) = digit.toByte
+         i += 1
+         digit >>= 8
+       }
+     }
+   }
+   bytes
+ }
+  */
+
+  def toByteArray(): Array[Byte] = {
+    if (this.sign == 0) {
+      return Array[Byte](0)
+    }
+    val temp: BigInteger = this
+    val bitLen = bitLength()
+    val firstNonZeroDigit = getFirstNonzeroDigit
+    var bytesLen = (bitLen >> 3) + 1
+    /*
+     * Puts the little-endian int array representing the magnitude of this
+     * BigInteger into the big-endian byte array.
+     */
+    val bytes = Array.ofDim[Byte](bytesLen)
+    var firstByteNumber = 0
+    var digitIndex = firstNonZeroDigit
+    var bytesInInteger = 4
+    var digit: Int = 0
+
+    val highBytes: Int =
```

Comment:
Put braces after `=` and at the end if the rhs of the assignment is not a one-liner nor a two-liner if/else.